### DISCUSSION TITLE
You application must have a secure (https) url in order to accept webhook calls

### DISCUSSION THREAD

**Discussion by spooksie** [def-studio/telegraph#103] [Q&A]
Hello, 

I get this error message when trying to create a webhook. I'm trying it out on local and my app is using valet with https. 

```
TelegramWebhookException

You application must have a secure (https) url in order to accept webhook calls
```

Any idea on why this is happening?

**Comment by fabio-ivona** [comment]
What's the resulting url from route `telegraph.webhook` ?

it should start with `https` in order to work

**Comment by pepo-ec** [comment]
I have started a bot from scratch and I have been following the documentation until trying to create the Webhook; **Could you tell me how I can use NGROK for the testing environment?** ... When I try to configure the webhook I have this message:

`php artisan telegraph:set-webhook `

>    DefStudio\Telegraph\Exceptions\TelegramWebhookException 
> 
>   You application must have a secure (https) url in order to accept webhook calls
> 
>   at vendor/defstudio/telegraph/src/Exceptions/TelegramWebhookException.php:24
>      20▕     }
>      21▕ 
>      22▕     public static function invalidScheme(): TelegramWebhookException
>      23▕     {
>   ➜  24▕         return new self("You application must have a secure (https) url in order to accept webhook calls");
>      25▕     }
>      26▕ }
>      27▕ 
> .
>       +16 vendor frames 
>   17  artisan:37
>       Illuminate\Foundation\Console\Kernel::handle()

**Reply by pepo-ec** [reply]
I have achieved it:

I installed Ngrok from the official site, I started it and then copied the HTTPS address in the APP_URL variable of _.env_

**Reply by fabio-ivona** [reply]
This is a good solution, happy you found your way!
